exporter/awsemfexporter/cwlog_client.go

```diff
@@ -175,9 +176,19 @@ func (client *cloudWatchLogClient) CreateStream(logGroup, streamName *string) (t
 	return "", nil
 }
 
-func newCollectorUserAgentHandler(buildInfo component.BuildInfo) request.NamedHandler {
+func newCollectorUserAgentHandler(buildInfo component.BuildInfo, logGroupName string) request.NamedHandler {
+	fn := request.MakeAddToUserAgentHandler(collectorDistribution, buildInfo.Version)
+	if matchContainerInsightsPattern(logGroupName) {
+		fn = request.MakeAddToUserAgentHandler(collectorDistribution, buildInfo.Version, "ContainerInsights")
+	}
 	return request.NamedHandler{
 		Name: "otel.collector.UserAgentHandler",
-		Fn:   request.MakeAddToUserAgentHandler(collectorDistribution, buildInfo.Version),
+		Fn:   fn,
 	}
 }
+
+func matchContainerInsightsPattern(logGroupName string) bool {
+	regexP := "^\\/aws\\/.*containerinsights\\/.*\\/(performance|prometheus)$"
+	r, _ := regexp.Compile(regexP)
```

Comment:
The compiled regex can be saved so that we don't need to compile it for each request.

Reply:
it's a good callout but this check will be only execute once per EMF Exporter instance. I think it is fine in this case. :)